### PR TITLE
Phase2 bug hunt version-1

### PR DESCRIPTION
**Pull request by lyazii22**
> Bug list

- [ ] Conflict between red exclamation mark next to the column name and green tick
in the "Issues and suggestions tab".
- [x] On specifying incorrect details for direct connection the loader keeps on loading even after clicking on home. Ideally that should stop once we click on home.
- [ ] Edits to the column names are not escaped.

### CODE REVIEW

File: ui/src/app/components/object-detail/object-detail.component.scss [hunk 1]
```diff
@@ -186,8 +186,8 @@
     flex-direction: row;
     justify-content: left;
     align-items: center;
-    .error {
-      color: #eb0000;
+    .note{
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.scss:189]: the icons are being shown as blue and its still showing error icon: https://drive.google.com/file/d/1fQM61eP_t43wnJd0nQMk7vurWp8gt88z/view?usp=sharing&resourcekey=0-M5NvbEfU-pgECOqYE56i7w

File: ui/src/app/services/interceptor/interceptor.service.ts [hunk 2]
```diff
@@ -10,8 +10,10 @@ export class InterceptorService implements HttpInterceptor {
   count: number = 0
   constructor(private loader: LoaderService) {}
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    this.loader.startLoader()
-    this.count++
+    if (req.url.includes('/connect')) {
```
Comment (shreyakhajanchi) [inline ui/src/app/services/interceptor/interceptor.service.ts:13]: the loader keeps on loading even after database is connected: https://drive.google.com/file/d/1fQM61eP_t43wnJd0nQMk7vurWp8gt88z/view?usp=sharing&resourcekey=0-M5NvbEfU-pgECOqYE56i7w

File: ui/src/app/components/object-detail/object-detail.component.scss (hunk 1 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.scss:189]: Initially we made it according to the figma design, but after the bug reported we made these changes, can you please suggest what icon can we show over there.

Comment (shreyakhajanchi) [reply ui/src/app/components/object-detail/object-detail.component.scss:189]: the bug meant using same icon not colour, in my opinion we can use warning icon in both the places, let me know if you have any better suggestion.

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 3]
```diff
@@ -202,6 +202,7 @@ <h3 class="title">
                   class="name_input"
                   type="text"
                   [formControl]="element.get('spColName')"
+                  onkeypress="return /[0-9a-zA-Z\_]/i.test(event.key)"
```
[the remark this answers is not in the document]
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.html:205]: Can you please elaborate a little more, a screenshot will be really helpful.

File: ui/src/app/services/interceptor/interceptor.service.ts (hunk 2 above)
Comment (lyazii22) [reply ui/src/app/services/interceptor/interceptor.service.ts]: Bug fixed

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.html]: Updated.

File: ui/src/app/components/object-detail/object-detail.component.scss (hunk 1 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.scss]: Updated

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 4]
```diff
@@ -49,6 +51,9 @@ export class ObjectDetailComponent implements OnInit {
         this.conv = res
       },
     })
+    this.userAddressValidations = this.formBuilder.group({
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.ts:54]: where is this used?

File: internal/report.go [hunk 5]
```diff
@@ -490,19 +490,19 @@ var IssueDB = map[SchemaIssue]struct {
 	Serial:                {Brief: "Spanner does not support autoincrementing types", severity: warning},
 	AutoIncrement:         {Brief: "Spanner does not support auto_increment attribute", severity: warning},
 	Timestamp:             {Brief: "Spanner timestamp is closer to PostgreSQL timestamptz", severity: suggestion, batch: true},
-	Datetime:              {Brief: "Spanner timestamp is closer to MySQL timestamp", severity: suggestion, batch: true},
-	Time:                  {Brief: "Spanner does not support time/year types", severity: note, batch: true},
-	Widened:               {Brief: "Some columns will consume more storage in Spanner", severity: note, batch: true},
+	Datetime:              {Brief: "Spanner timestamp is closer to MySQL timestamp", severity: warning, batch: true},
```
Comment (shreyakhajanchi) [inline internal/report.go:493]: this is a suggestion as marked in: https://docs.google.com/spreadsheets/d/1qx-gS4t0JNki76_bkp-Iag-9cW3MBtfJb2vn7hvm82Y/edit?hl=en&forcehl=1#gid=0

File: ui/src/app/components/summary/summary.component.html [hunk 6]
```diff
@@ -76,15 +76,15 @@
               *ngIf="item.type == 'suggestion'"
               class="suggestion"
             >
-              check_circle
+            wb_incandescent
```
Comment (shreyakhajanchi) [inline ui/src/app/components/summary/summary.component.html:79]: The suggestion icon appears to be inverted: https://drive.google.com/file/d/1Uq3ETl1DJ2OPn6lF8gmv2DM0JZzroQLQ/view?usp=sharing&resourcekey=0-fPPXeGS4g8TBjTn8Ig9fdA

Comment (Ronak-Searce) [reply ui/src/app/components/summary/summary.component.html:79]: Updated

File: ui/src/app/components/object-detail/object-detail.component.ts (hunk 4 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.ts]: Updated, it was supposed to be removed.

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**